src/Scaffolding/Scaffolders/DataObjectScaffolder.php

```
@@ -0,0 +1,338 @@
+
+        return $this;
+    }
+
+    /**
+     * Adds visible fields.
+     * @param array $fields
+     */
+    public function addFields(array $fields)
+    {
+        // Remove duplicates
+        $this->fields = ArrayList::create(array_unique(array_merge(
+            $this->fields->toArray(),
+            $fields
+        )));
+
+        return $this;
+    }
+
+    /**
+     * @param $field
+     * @return mixed
+     */
+    public function addField($field)
+    {
+        return $this->addFields((array)$field);
+    }
+
+    /**
+     * @param $field
+     * @return $this
+     */
+    public function removeField($field)
+    {
+        $this->fields->remove($field);
+
+        return $this;
+    }
+
+    /**
+     * @param array $fields
+     * @return $this
+     */
+    public function removeFields(array $fields)
+    {
+        foreach ($fields as $field) {
+            $this->removeField($field);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function getQueries()
+    {
+        return $this->queries;
+    }
+
+    /**
+     * @return array
+     */
+    public function getMutations()
+    {
+        return $this->mutations;
+    }
+
+    /**
+     * @return array
+     */
+    public function getFields()
+    {
+        return $this->fields;
+    }
+
+
+    /**
+     * Puts all of the configuration into a type creator
+     * @return DataObjectTypeCreator
+     */
+    public function getCreator(Manager $manager)
+    {
+        $fieldMap = [];
+        $instance = $this->getDataObjectInstance();
+        $extraDataObjects = $this->getExtraDataObjects();
+        $fields = array_unique($this->fields->toArray());
+
+        if (empty($fields)) {
+        	$fields = Config::inst()->get(self::class, 'default_fields');        	
+        }
+
+        foreach ($fields as $fieldName) {
+            $result = $instance->obj($fieldName);
+            if ($result instanceof DBField) {
```

How do we deal with getters? Force $casting? Then we should throw an exception if the result is not an instance of DBField